### PR TITLE
linode networking view

### LINKED ISSUE #34: Implement Linode detail networking tab (view only)
This only entails viewing the information here. Note that feedback from @caker suggests we do not hide any IPs but instead just show the full list at all times.

### PR DESCRIPTION
**Pull request by na3d**
Closes #34 
screen shot 2016-06-23 at 9 52 38 am

### CODE REVIEW

File: src/linodes/layouts/LinodeNetworking.js [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+  dallas: '2600:3c00::',
+  fremont: '2600:3c01::',
+  atlanta: '2600:3c02::',
+  newark: '2600:3c03::',
+  london: '2a01:7e00::',
+  tokyo: '2400:8900::',
+  singapore: '2400:8901::',
+  frankfort: '2a01:7e01::',
+};
+export const ipv6nsSuffix = ['5', '6', '7', '8', '9', 'b', 'c'];
```
Comment (SirCmpwn) [inline src/linodes/layouts/LinodeNetworking.js]: Can you move these into constants.js?

File: src/linodes/layouts/LinodeNetworking.js [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+
+function ipList(ips) {
+  const list = [];
+
+  ips.forEach(
+    ip => list.push(<li key={ip}>{ip}</li>)
+  );
+
+  return list;
+}
```
Comment (SirCmpwn) [inline src/linodes/layouts/LinodeNetworking.js]: function ipList(ips) {
        return ips.map(ip => <li key={ip}>{ip}</li>);
    }

Or better yet just inline it

File: src/linodes/layouts/LinodeNetworking.js [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+      </div>
+    </div>
+  );
+
+  return nameservers;
+}
+
+export class LinodeNetworking extends Component {
+
+  constructor() {
```
Comment (SirCmpwn) [inline src/linodes/layouts/LinodeNetworking.js]: Extra whitespace, nuke it

File: src/linodes/layouts/LinodeNetworking.js [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+    super();
+    this.getLinode = getLinode.bind(this);
+  }
+
+  render() {
+    /* To Do
```
Comment (SirCmpwn) [inline src/linodes/layouts/LinodeNetworking.js]: Make this TODO so it mathces the others, easier to grep

File: src/linodes/layouts/LinodeNetworking.js [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+          <div className="col-sm-4 linode-label-col left">
+            Gateway
+          </div>
+          <div className="col-sm-8 linode-content-col right">
+            {ipv4[0].substring(0, ipv4[0].lastIndexOf('.'))}.1
+          </div>
+        </div>
+        {nameserversList(true, linode)}
+      </div>
+    );
```
Comment (SirCmpwn) [inline src/linodes/layouts/LinodeNetworking.js]: How about splitting this up into a bunch of render* helper functions? i.e. `{this.renderIPv4Public()}`

File: src/linodes/layouts/LinodeNetworking.js [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+              suffix => ipv6ns[dc] + suffix
+            ))
+          }
+        </ul>
+      </div>
+    </div>
+  );
+
+  return nameservers;
+}
```
Comment (SirCmpwn) [inline src/linodes/layouts/LinodeNetworking.js]: I'd move this into the class, too.

File: test/data.js [hunk 7]
```diff
@@ -26,6 +26,7 @@ export const testLinode = {
   datacenter: {
     id: 'datacenter_6',
     label: 'Newark, NJ',
+    datacenter: 'newark',
```
Comment (SirCmpwn) [inline test/data.js]: ...is this a thing?

Comment (SirCmpwn) [inline test/data.js]: I guess it is. Neat.

File: test/linodes/layouts/LinodeNetworking.spec.js [hunk 8]
```diff
@@ -0,0 +1,254 @@
...
+          },
+          private: {
+            ipv4: [
+              '321.456.789.1',
+            ],
+            link_local: 'fe80::f03c:91ff:fe96:0f13',
+          },
+        },
+      },
+    },
```
Comment (SirCmpwn) [inline test/linodes/layouts/LinodeNetworking.spec.js]: Considered adding IPs to the basic testLinode? It's meant to represent a fully capable Linode object for testing purposes so you don't have to define one here.

File: src/linodes/layouts/LinodeNetworking.js [hunk 9]
```diff
@@ -0,0 +1,310 @@
...
+              <div className="input-group-btn">
+                <button type="button" id="glish-button" className="btn btn-default pull-right">
+                  Add IP address
+                </button>
+              </div>
+            </div>
+          </div>
+          <div className="row network-content">
+              {ipv4Public}
+              {ipv6Public}
```
Comment (eatonphil) [inline src/linodes/layouts/LinodeNetworking.js]: How does the linter let this go? Isn't it supposed to be two spaces?

File: test/data.js (hunk 7 above)
Comment (na3d) [inline test/data.js]: Yeah, had to make it for the general tab for Lish.

**Comment by eatonphil** [comment]
The helper link on the right of the ipv4 and ipv6 sections is a problem. Perhaps it should be absolutely positioned into the margin. Or maybe it should just get lifted out and put to the right of the Public section similar to the Private section.

File: test/linodes/layouts/LinodeNetworking.spec.js (hunk 8 above)
Comment (na3d) [inline test/linodes/layouts/LinodeNetworking.spec.js]: Ok, I will swap that and the private IP tests.

**Comment by na3d** [comment]
screen shot 2016-06-23 at 1 57 11 pm

**Comment by eatonphil** [comment]
Looks like you've got some extra random whitespace after the Public section. And it looks like the padding above and below Public network is weird (equal). Is this based off master?

**Comment by na3d** [comment]
It is just padding from the line, we can easily adjust it as desired.

**Comment by eatonphil** [comment]
Use $main-padding-tb for the padding above and below the line and have no other padding other than that.

**Comment by na3d** [comment]
screen shot 2016-06-23 at 1 57 11 pm

**Comment by na3d** [comment]
screen shot 2016-06-23 at 2 26 34 pm

File: scss/layout/linode-details.scss [hunk 10]
```diff
@@ -83,6 +83,11 @@
         }
     }
 
+    hr {
+        margin-top: $main-padding-tb;
+        margin-bottom: $main-padding-tb;
```
Comment (eatonphil) [inline scss/layout/linode-details.scss]: margin: $main-padding-tb 0;

Comment (eatonphil) [inline scss/layout/linode-details.scss]: Ok if you could also add:

    .row:last-of-type > .linode-content-col {
        margin-bottom: 0;
    }

    .graphs {
        margin-top: $main-padding-tb;
    }

That would be perfect.

**Comment by na3d** [comment]
screen shot 2016-06-23 at 3 00 40 pm

**Comment by na3d** [comment]
screen shot 2016-06-23 at 3 32 37 pm

**Comment by na3d** [comment]
screen shot 2016-06-23 at 3 44 23 pm

**Comment by na3d** [comment]
screen shot 2016-06-24 at 8 43 32 am

**Comment by eatonphil** [comment]
Ok! Great, I'm happy with this now. Ready for merge?

**Comment by na3d** [comment]
👍

**Pull request merged**